server/src/main/java/org/apache/cassandra/sidecar/routes/AccessProtectedRouteBuilder.java

```diff
@@ -0,0 +1,208 @@
+        return this;
+    }
+
+    /**
+     * Sets if BodyHandler should be created for the route.
+     *
+     * @param setBodyHandler boolean flag indicating if route requires BodyHandler
+     * @return a reference to {@link AccessProtectedRouteBuilder} for chaining
+     */
+    public AccessProtectedRouteBuilder setBodyHandler(Boolean setBodyHandler)
+    {
+        this.setBodyHandler = setBodyHandler;
+        return this;
+    }
+
+    /**
+     * Adds handler to handler chain of route. Handlers are ordered, they are called in order they are set in chain.
+     *
+     * @param handler handler for route
+     * @return a reference to {@link AccessProtectedRouteBuilder} for chaining
+     */
+    public AccessProtectedRouteBuilder handler(Handler<RoutingContext> handler)
+    {
+        this.handlers.add(handler);
+        return this;
+    }
+
+    /**
+     * Builds an authorized route. Adds {@link io.vertx.ext.web.handler.AuthorizationHandler} at top of handler
+     * chain if access control is enabled.
+     */
+    public void build()
+    {
+        Preconditions.checkArgument(router != null, "Router must be set");
+        Preconditions.checkArgument(method != null, "Http method must be set");
+        Preconditions.checkArgument(endpoint != null && !endpoint.isEmpty(), "Endpoint must be set");
+        Preconditions.checkArgument(!handlers.isEmpty(), "Handler chain can not be empty");
+
+        Route route = router.route(method, endpoint);
+
+        // BodyHandler should be at index 0 in handler chain
+        if (setBodyHandler)
+        {
+            route.handler(BodyHandler.create());
+        }
+
+        if (accessControlConfiguration.enabled())
+        {
+            // authorization handler added before route specific handler chain
+            AuthorizationWithAdminBypassHandler authorizationHandler
+            = new AuthorizationWithAdminBypassHandler(adminIdentityResolver, requiredAuthorization());
+            authorizationHandler.addAuthorizationProvider(authorizationProvider);
+            authorizationHandler.variableConsumer(routeGenericVariableConsumer());
+
+            route.handler(authorizationHandler);
+        }
+        handlers.forEach(route::handler);
+    }
+
+    private Authorization requiredAuthorization()
+    {
+        Set<Authorization> requiredAuthorizations = handlers
+                                                    .stream()
+                                                    .filter(handler -> handler instanceof AccessProtected)
+                                                    .map(handler -> (AccessProtected) handler)
+                                                    .flatMap(handler -> handler.requiredAuthorizations().stream())
+                                                    .collect(Collectors.toSet());
+        if (requiredAuthorizations.isEmpty())
+        {
+            throw new ConfigurationException("Authorized route must have enforced authorizations set");
```

Comment:
Meant to say that authorizations should have been 'declared'? If yes, then does it sound better?

```suggestion
            throw new ConfigurationException("Authorized route must have authorizations declared");
```